libbpfgo.go

```diff
@@ -440,6 +440,37 @@ func (b *BPFMap) GetMaxEntries() uint32 {
 	return uint32(maxEntries)
 }
 
+func (b *BPFMap) GetFd() int {
+	return int(b.fd)
+}
+
+func (b *BPFMap) GetName() string {
+	return b.name
+}
+
+func (b *BPFMap) GetModule() *Module {
+	return b.module
+}
+
+func (b *BPFMap) GetPinPath() string {
+	cs := C.CString(b.name)
+	bpfMap := C.bpf_object__find_map_by_name(b.module.obj, cs)
```

Comment:
This step is unnecessary as you can get it from the bpfMap field of a BPFMap

Reply:
Yes we can get it inside the BPFMap, indeed. I've noticed that other functions do the same, getting again the same reference:

https://github.com/aquasecurity/libbpfgo/blob/b1eb1b9a6b20b60d0cc0ebfb374ec08684d1682e/libbpfgo.go#L380-L383

https://github.com/aquasecurity/libbpfgo/blob/b1eb1b9a6b20b60d0cc0ebfb374ec08684d1682e/libbpfgo.go#L392-L395

And so on.

Is there a specific reason for those?

Reply:
We should change these as well.

> Would it be related to maps that could be changed by other codepaths

Changed how? bpfMap (bpf_map) is used by libbpf as a reference to a map so that it can use the bpf syscall with. It is mostly the fd of the struct that is used for these operations. I don't see a problem with working with that for the lifecycle of the BPFMap object.

Reply:
> We should change these as well.

Grand, I can do that in this PR or a separated one. Please let me know what you do prefer.

> Changed how?

Other software using direct access (bpf syscall) could pin or unpin or set a different pinpath to one running map. Would be the libbpf maps fields automatically updated in this case?

Reply:
> Grand, I can do that in this PR or a separated one. Please let me know what you do prefer.

You can add it to this PR if you like, not a must though.

> Other software using direct access (bpf syscall) could pin or unpin or set a different pinpath to one running map. Would be the libbpf maps fields automatically updated in this case?

Even when other programs pin/unpin/change pin path, as long as you have a reference to a map, you can use it to do other operations on it. Here is the code in libbpf that is being called from here: https://github.com/libbpf/libbpf/blob/master/src/libbpf.c#L8509 - as you can see, the bpfMap reference is enough to get the relevant data. So in our context, I don't think it is a problem (although you are right that it may be a problem for the validity of some other fields, like if the map is pinned or not, but we don't care about it here).

Reply:
> You can add it to this PR if you like, not a must though.

Done. Anything, please let me know. :+1:

Reply:
Nice catch on bpfMap. All tests seem fine, I approved it. @yanivagman let me know if you're good so I can merge, or feel free to merge it yourself once u think is good. Thanks!

Reply:
I'll go ahead and merge this if you don't mind, we can always change it if needed =).